tests/WorkerTest.php

```diff
@@ -114,12 +91,16 @@ public function runNextJob_EntityManagerClosedAndJobFailed_JobNotRequeuedAndWork
     public function runNextJob_DatabaseConnectionClosed_ShouldReOpenConnection(): void
     {
         // Assemble
-        $this->entityManager->shouldReceive('isOpen')->andReturn(true);
-        $this->entityManager->shouldReceive('clear');
+        $this->entityManager->shouldReceive('getConnection')->andReturn($this->connection)->once();
+        $this->entityManager->shouldReceive('isOpen')->andReturn(true)->once();
+        $this->entityManager->shouldReceive('clear')->once();
 
-        $this->connection->shouldReceive('ping')->andReturn(false);
-        $this->connection->shouldReceive('close');
-        $this->connection->shouldReceive('connect');
+        $this->platform->shouldReceive('getDummySelectSQL')->andReturn('SELECT 1')->once();
+
+        $this->connection->shouldReceive('executeQuery')->once();
+        $this->connection->shouldReceive('getDatabasePlatform')->andReturn($this->platform)->once();
+        $this->connection->shouldNotReceive('close');
+        $this->connection->shouldNotReceive('connect');
```

Comment:
How come these changed from `shouldReceive()` to `shouldNotReceive()`? How are we verifying that the worker reopened the connection when it's closed? Shouldn't the call to `executeQuery` throw a `ConnectionLost` exception? (assuming that's the exception thrown when trying to execute a query on a closed database connection).

Reply:
Whoops, my mistake. This was the function I was testing the `ping()` replacement fix on.
Nice catch!